js/app.js

```diff
@@ -0,0 +1,263 @@
+    wKeyPressed = true;        
+  }
+  // Allow the car to turn only when moving fordwards or backwards
+  if(event.keyCode === 83 && (dKeyPressed || aKeyPressed)) {
+    sKeyPressed  = true;        
+  }
+  
+  // Pause game
+  if (event.keyCode === 80) {
+    raceStarted = ! raceStarted;
+  }
+}
+
+function keyUpHandler(event) {
+  if(event.keyCode === 39) {
+    rightKeyPressed = false;        
+  }
+
+  if(event.keyCode === 37) {
+    leftKeyPressed  = false;        
+  }
+  
+  if(event.keyCode === 38) {
+    upKeyPressed = false;        
+  }
+
+  if(event.keyCode === 40) {
+    downKeyPressed  = false;        
+  }
+  
+  if(event.keyCode === 68) {
+    dKeyPressed = false;        
+  }
+
+  if(event.keyCode === 65) {
+    aKeyPressed  = false;        
+  }
+  
+  if(event.keyCode === 87) {
+    wKeyPressed = false;        
+  }
+
+  if(event.keyCode === 83) {
+    sKeyPressed  = false;        
+  }
+}
+
+
+function drawCarOne(x, y) {
+  var carOneImage = new Image();
+  carOneImage.src="images/player-one-car.png";
+  ctx.drawImage(carOneImage, x, y);
+}
+
+
+function drawCarTwo(x, y) {
+  var carTwoImage = new Image();
+  carTwoImage.src="images/player-two-car.png";
+  ctx.drawImage(carTwoImage, x, y);
+}
+
+
+function drawRaceTrack() {
+  ctx.beginPath();
+  ctx.rect(TRACK_X, TRACK_Y, TRACK_WIDTH, TRACK_HEIGHT);
+  ctx.fillStyle = "#4f4f4f";
+  ctx.fill();
+  ctx.closePath();
+  for(var i=0, l=15; i<l; i++) {
+    ctx.beginPath();
+    ctx.rect(100 * i, 2 * TRACK_Y, 70, 10);
+    ctx.fillStyle = "white";
+    ctx.fill();
+    ctx.closePath();
+  }
+  // Finish line
+  ctx.beginPath();
+  ctx.moveTo(canvas.width - CAR_WIDTH - 20, TRACK_Y);
+  ctx.lineTo(canvas.width - CAR_WIDTH - 20, TRACK_Y + TRACK_HEIGHT);
+  ctx.lineWidth = 5;
+  ctx.strokeStyle = "red";
+  ctx.stroke();
+  ctx.closePath();
+  
+  // Start line
+  ctx.beginPath();
+  ctx.moveTo(TRACK_X + 10 + CAR_WIDTH, TRACK_Y);
+  ctx.lineTo(TRACK_X + 10 + CAR_WIDTH, TRACK_Y + TRACK_HEIGHT);
+  ctx.lineWidth = 5;
+  ctx.strokeStyle = "green";
+  ctx.stroke();
+  ctx.closePath();
+}
+
+// Detect collision between cars by checking
+// if the rectangles representing both cars overlap
+function detectCollision() {
+  if (carOne.x < carTwo.x + CAR_WIDTH - 10 &&
+      carOne.x + CAR_WIDTH - 10 > carTwo.x &&
+      carOne.y < carTwo.y + CAR_HEIGHT - 10 &&
+      CAR_HEIGHT - 10 + carOne.y > carTwo.y) {
+    alert('Cars met with an accident. Starting a new race.');
+    document.location.reload(); 
+  }
+}
+
+function drawPauseText() {
+  ctx.font="45px Arial";
+  ctx.fillStyle = "#E53935";
+  ctx.textAlign = "center";
+  ctx.fillText("Game Paused. Press 'P' to resume." , canvas.width/2, canvas.height/2);
+}
+
+// Render all game components
+function renderGame() {
+  // Clear canvas before rendering
+  ctx.clearRect(0, 0, canvas.width, canvas.height);
+  // Draw components
+  drawRaceTrack();
+  drawCarOne(carOne.x, carOne.y);
+  drawCarTwo(carTwo.x, carTwo.y);

```

Comment:
Given that `renderGame` is called in a loop, do you foresee any issues with calling through to `drawCarOne`, which instantiates a new image (line 113) each time?